uaclient/tests/test_security.py

```diff
@@ -1479,17 +1485,20 @@ def fake_attach(args, cfg):
         m_action_attach.side_effect = fake_attach
 
         cfg = FakeConfig()
-        with mock.patch("uaclient.util.sys") as m_sys:
-            m_stdout = mock.MagicMock()
-            type(m_sys).stdout = m_stdout
-            type(m_stdout).encoding = mock.PropertyMock(return_value="utf-8")
-            prompt_for_affected_packages(
-                cfg=cfg,
-                issue_id="USN-###",
-                affected_pkg_status=affected_pkg_status,
-                installed_packages=installed_packages,
-                usn_released_pkgs=usn_released_pkgs,
-            )
+        with mock.patch("uaclient.util._subp", side_effect=_subp):
```

Comment:
Just out of curiosity, what is the `subp`  command that we need to test the `prompt_for_affected_packages` ?
Because for that command, I think the test needs to be mocking all of those calls

Reply:
`version_cmp_le` uses `dpkg --compare-versions` and is important to the functionality of`prompt_for_affected_packages`. There are probably some other `subp`s slipping through here though :/. But the "correct" fix would probably be to re-write `version_cmd_le` in python instead of relying on dpkg and that didn't seem in-scope for this right now.